frontend/src/pages/register.js

```diff
@@ -12,8 +12,12 @@ import { register } from "../store/user";
 import InputField from "../components/input";
 import { validate, validateField, getErrorPayload } from "../utils/validation";
 import * as userService from "../services/userService";
+import getCookie from "../utils/getCookie";
 
 const Register = () => {
+  const isAuthenticated = getCookie("isAuthenticated");
+  const navigate = useNavigate();
+  if (isAuthenticated) navigate("/home");
```

Comment:
Your authentication indicator is so messy.  Can be found here and there. Repetitive code on every page.

Remember the DRY concept. Dont repeat yourself. If you find a repetitive code, find a way to refactor it.

In this case, make a auth context provider. 
Important thing here is the useAuth.js file and the hook below then import it in the app.js
```
useEffect(() => {
    usersApi.getCurrentUser()
      .then((user) => setUser(user))
      .catch((_error) => {})
      .finally(() => setLoadingInitial(false));
  }, []);
```
https://dev.to/finiam/predictable-react-authentication-with-the-context-api-g10